core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentProperties.java

```diff
@@ -287,31 +287,31 @@ private void fillDimensionAndMeasureDetails(List<ColumnSchema> columnsInTable,
           // if it is a columnar dimension participated in mdkey then added
           // key ordinal and dimension ordinal
           carbonDimension =
-              new CarbonDimension(columnSchema, dimensonOrdinal++, keyOrdinal++, -1);
+              new CarbonDimension(columnSchema, dimensionOrdinal++, keyOrdinal++, -1);
         }
         // as complex type will be stored at last so once complex type started all the dimension
         // will be added to complex type
         else if (isComplexDimensionStarted || columnSchema.getDataType().isComplexType()) {
           cardinalityIndexForComplexDimensionColumn.add(tableOrdinal);
           carbonDimension =
-              new CarbonDimension(columnSchema, dimensonOrdinal++, -1, ++complexTypeOrdinal);
+              new CarbonDimension(columnSchema, dimensionOrdinal++, -1, ++complexTypeOrdinal);
           carbonDimension.initializeChildDimensionsList(columnSchema.getNumberOfChild());
           complexDimensions.add(carbonDimension);
           isComplexDimensionStarted = true;
-          int previouseOrdinal = dimensonOrdinal;
-          dimensonOrdinal =
-              readAllComplexTypeChildren(dimensonOrdinal, columnSchema.getNumberOfChild(),
+          int previouseOrdinal = dimensionOrdinal;
```

Comment:
um, part of the variable name 'previouse' is also typo, which should be 'previous'...
Please correct this also and others LGTM.